src/DeterministicFactory_v1.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.23;
+
+// Internal Interfaces
+import {IDeterministicFactory_v1} from
+    "src/interfaces/IDeterministicFactory_v1.sol";
+
+// External Dependencies
+import {Ownable2Step, Ownable} from "@oz/access/Ownable2Step.sol";
+import {Create2} from "@oz/utils/Create2.sol";
+
+/**
+ * @title   Deterministic Deployment Factory
+ *
+ * @notice  This contract is a factory for deploying contracts using the CREATE2 opcode,
+ *             allowing us to deploy some of our contract at the same address of each EVM-
+ *          compatible network.
+ *
+ * @dev     We limit this to a specific address ('allowedDeployer'), as otherwise anyone
+ *          could deploy our contracts at the addresses we use. This can be problematic
+ *          for contracts that are initialized, i.e. where the owner is only set
+ *          after deployment, which would hand them ownership of the contracts. We intend
+ *          to set this 'allowedDeployer' to the deployment account we use, and then
+ *          set it to the zero address afterwards (until the next deployment takes place).
```

Comment:
This will be done in the Safe UI. As the contract will be verified, we don't need a script for this for mainnets. I can still add one for testing purposes though, it's a good idea.

(And yes, that is part of the guide document)